server/middlewares/auth.js

```diff
@@ -1,6 +1,45 @@
-const auth = (req, res, next) => {
-    // placeholder auth middleware
-    return next();
+require("dotenv").config();
+const { verify_jwt } = require("./verify_jwt");
+const axios = require("axios");
+
+const auth = async (req, res, next) => {
+    // make it possible to skip actual auth for testing/development
+    var skip = process.env.SKIP_AUTH === "true";
+    if (skip) {
+        return next();
+    }
+
+    const access_token = req.header("accessToken");
+    const refresh_token = req.header("refreshToken");
+
+    if (!access_token || !refresh_token) {
+        return res.status(403).send("missing token");
+    }
+    const { payload, expired } = verify_jwt(access_token);
+
+    if (payload && !expired) {
+        req.user = payload;
+        return next();
+    }
+
+    if (expired) {
+        await axios
+            .post(`${process.env.MAIN_HUB_HOST}/api/token`, {
+                refreshToken: refresh_token,
```

Comment:
Ist der Header mit dem MainHub-Team abgesprochen?

Reply:
weiß nicht was du meinst

Reply:
wenn du den body da meinst dann hab ich das gefixed. 
die ziehen sich da halt aus dem body `const refreshToken = req.body.token`.
hab refreshToken in token umbenannt

Reply:
My bad.